decide/lic4.test.cpp

```diff
@@ -0,0 +1,69 @@
+#include <cassert>
+#include <cstdlib>
+#include <decide/decide.hpp>
+#include <decide/lics.hpp>
+
+
+
+//test that when QUADS = 1 then it will be true
+// this is because any point will at least be in one quad
+bool test1_true(){
+    Points points{Coordinate(1, 1), Coordinate(0, -1), Coordinate(0, 2),Coordinate(-5, -4), Coordinate(1, -9), Coordinate(0, 0)};
+    Parameters param{.QPTS=3, .QUADS=1};
+    assert(lic4(points,param));
+}
+//the last four are different 
+bool test2_true(){
+    Points points{Coordinate(1, 1), Coordinate(0, 7), Coordinate(16, 9),Coordinate(-8, 4), Coordinate(1, -9), Coordinate(-7, -19)};
+    Parameters param{.QPTS=4, .QUADS=3};
+    assert(lic4(points,param));
+}
+
+//checking that it works if it is only looking for 3 quads
+bool test3_true(){
+    Points points{Coordinate(1, 1), Coordinate(0, -1), Coordinate(0, 2),Coordinate(-5, -4), Coordinate(1, -9), Coordinate(0, 0)};
+    Parameters param{.QPTS=6, .QUADS=2};
+    assert(lic4(points,param));
+}
+
+//should be false
+//all points are the same so they can't be in different quads
+bool test4_false(){
+    Points points{Coordinate(1, 1), Coordinate(1, 1), Coordinate(1, 1),Coordinate(1, 1), Coordinate(1, 1), Coordinate(1, 1)};
+    Parameters param{.QPTS=3, .QUADS=3};
+    assert(lic4(points,param));
```

Comment:
Should assert that it will be false, i.e. `assert(!lic4(points,param));` (adds an exclamation mark)

This applies for all the other tests here that should return false.